pkg/server/server.go

```diff
@@ -184,3 +185,27 @@ func getEncodedContent(inp string) string {
 		Opaque: "," + dataurl.Escape([]byte(inp)),
 	}).String()
 }
+
+// MigrateKernelArgsIfNecessary moves the kernel arguments back into MachineConfig when going from a version that supports
+// ignition kernel arguments to a version that does not. Without this, we would be unable to serve anything < 3.3 because the
+// ignition converter fails to downconvert if unsupported fields are populated. If ShouldNotExist is populated in the
+// ignition kernel args, we still have to fail because there is nowhere in MachineConfig for us to put those.
+func MigrateKernelArgsIfNecessary(conf *ign3types.Config, mc *mcfgv1.MachineConfig, version *semver.Version) error {
+	// If we're downgrading from a version of ignition that
+	// support kargs, we need to stuff them back into MachineConfig so they still end up in the encapsulation
+	// and they still get applied
+	if version != nil && version.LessThan(*semver.New("3.3.0")) {
+		// we're converting to a version that doesn't support kargs, so we need to stuff them in machineconfig
+		if len(conf.KernelArguments.ShouldNotExist) > 0 {
+			return fmt.Errorf("Can't serve version %s with ignition KernelArguments.ShouldNotExist populated", version)
```

Comment:
version converts to string?

Reply:
Yep, satisfies stringer interface:  https://github.com/coreos/go-semver/blob/6328231d806b2e2199c42ecc055a16ddcc2c6121/semver/semver.go#L106